File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaModuleRepository.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories;

import java.util.UUID;

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.ModuleEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface JpaModuleRepository extends JpaRepository<ModuleEntity, UUID> {
    ModuleEntity findByCode(String code);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/repositories/JpaModuleRepository.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.repositories

import br.com.deltaglobalbank.identity.infrastructure.persistence.entities.ModuleEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface JpaModuleRepository : JpaRepository<ModuleEntity, UUID> {
    fun findByCode(code: String): ModuleEntity?
}
